lib/e2e/src/error.rs

```diff
@@ -92,3 +95,22 @@ impl<E: SolError> Revert<E> for alloy::contract::Error {
         expected == actual
     }
 }
+
+impl<E: SolError> Revert<E> for eyre::Report {
+    fn reverted_with(&self, expected: E) -> bool {
+        let Some(received) = self
+            .chain()
+            .find_map(|err| err.downcast_ref::<RpcError<TransportErrorKind>>())
+        else {
+            return false;
+        };
+        let RpcError::ErrorResp(received) = received else {
+            return false;
+        };
+        let Some(received) = &received.data else {
+            return false;
+        };
+        let expected = alloy::hex::encode(expected.abi_encode());
+        received.to_string().contains(&expected)
+    }
+}
```

Comment:
@alexfertel if this approach for constructor error checks is fine for you. We do not need to make additional improvements to koba error types and publish new release to crates.io
Also I've realized that the new version of koba should be `0.2.0` and not `0.1.3` since recent changes were compatibility breaking.

Reply:
Yeah, this looks good to me. We should be able to simplify it a bit in the future once a few changes are published in `alloy`, but I like that we don't need to complicate bulloak's API too much 👍